src/ui/win/win_ui.cpp

```diff
@@ -261,6 +274,27 @@ void UiWindowWin::drawLine( const Point2d & start, const Point2d & end, const Pa
     _display();
 }
 
+void UiWindowWin::drawEllipse( const Point2d & center, double xRadius, double yRadius, const PaintColor & color )
+{
+    _ellipse.emplace_back( center.x - xRadius, center.y - yRadius, center.x + xRadius, center.y + yRadius, color );
+
+    _display();
+}
+
+void UiWindowWin::drawRectangle( const Point2d & topLeftCorner, double width, double height, const PaintColor & color )
+{
+    const Point2d topRightCorner = Point2d( topLeftCorner.x + width, topLeftCorner.y );
```

Comment:
You can use a constructor here instead of assign operator. **const Point2d name(x, y);**

Reply:
I'm not sure I get what you mean.

Reply:
I'll make changes. You'll see :)